qiskit/transpiler/passes/utils/gate_direction.py

```diff
@@ -277,11 +285,21 @@ def _run_target(self, dag, wire_map):
                         f"The circuit requires a connection between physical qubits {qargs}"
                         f" for {node.name}"
                     )
-            else:
+            elif self.target.instruction_supported(node.name, qargs):
+                continue
+            elif self.target.instruction_supported(node.name, swapped) or dag.has_calibration_for(
+                _swap_node_qargs(node)
+            ):
                 raise TranspilerError(
                     f"Flipping of gate direction is only supported "
-                    f"for {list(self._static_replacements)} at this time, not '{node.name}'."
+                    f"for {list(self._KNOWN_REPLACEMENTS)} at this time, not '{node.name}'."
                 )
+            else:
+                raise TranspilerError(
+                    f"'{node.name}' with parameters '{node.op.params}' is not supported on qubits"
```

Comment:
This message could seem bit weird when parameter is empty.

Reply:
I guess _marginally_ weird, but still perfectly understandable, right?  It'll just say something like

> 'cx' with parameters '[]' is not supported

which still looks pretty sensible and easy to parse.

Reply:
I can change it if you'd prefer, though.

Reply:
The point is that a `Target` can allow a gate with certain parameters on certain qubit pairs, but not other parameter values or qubits.  I put the parameters in because I didn't want people to be confused if their target supports (say) `my_rotation` with a specific angle of $\pi/2$ but not other angles, and the error just says "my_rotation isn't supported".

I think it's a bit beyond scope for us to try and pull out all "similar" instructions from the target and try and guess which of those the user might have meant, though.

Reply:
Fair enough.